osu.Game/Database/IRealmFactory.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using Realms;
+
+namespace osu.Game.Database
+{
+    public interface IRealmFactory
+    {
+        /// <summary>
+        /// The main realm context, bound to the update thread.
+        /// </summary>
+        public Realm Context { get; }
```

Comment:
Also is there a reason why everything can't go through `GetForRead`/`GetForWrite`, even if it's on the update thread? This being exposed makes me a bit worried about misuse.

Reply:
I guess that would mean adding a conditional case to `GetForRead` which returned a custom `RealmUsage` that used the update thread realm and didn't clean up the context after usage? Is a potential direction we could try. I thought this way it's very deliberate when you use the update thread realm context, vs getting a temporary/local one.

Reply:
Is there a case where you'd care about which thread the context is being retrieved on? Is the tradeoff in security - the fact that you can access this off-thread and potentially blow everything up - worth it?

Reply:
A case in point is the usage in `ToolbarButton`, which keeps a reference to a live realm model (and consumes it on the update thread). In such a usage, it feels a bit weird to be disposing a `RealmUsage` in my opinion, since that operation suggests that it would be going out of context.

Reply:
I see...